Added the fix for NPE happening during merges when all vector fields docs are deleted in the segments getting merged (#2365)
opensearch-project/k-NN: CHANGELOG.md

- Add check to directly use ANN Search when filters match all docs. (#2320)[https://github.com/opensearch-project/k-NN/pull/2320]
### Bug Fixes
* Fixing the bug when a segment has no vector field present for disk based vector search (#2282)[https://github.com/opensearch-project/k-NN/pull/2282]

> * Added the fix for NPE happening during merges when all vector fields docs are deleted in the segments getting merged (#2365)[https://github.com/opensearch-project/k-NN/pull/2365]

Reviewer: ```suggestion
*  Fix for NPE while merging segments after all the vector fields docs are deleted (#2365)[https://github.com/opensearch-project/k-NN/pull/2365]
```

Author: fixed
